libraries/matrix/api/src/main/kotlin/io/element/android/libraries/matrix/api/exception/ClientException.kt

```diff
@@ -7,10 +7,10 @@
 
 package io.element.android.libraries.matrix.api.exception
 
-sealed class ClientException(message: String) : Exception(message) {
-    class Generic(message: String) : ClientException(message)
-    class MatrixApi(val kind: ErrorKind, val code: String, message: String) : ClientException(message)
-    class Other(message: String) : ClientException(message)
+sealed class ClientException(message: String, val details: String?) : Exception(message) {
```

Comment:
I am wondering if we should exploit this `details` at the application level.

Maybe something like:

```kotlin
sealed class ClientException(message: String, details: String?) : Exception(
    buildString {
        append(message)
        if (details != null) {
            append(" (")
            append(details)
            append(")")
        }
    }
) {
```

Not a blocker though.